resources/js/components/ui/Capitalize.tsx

```diff
@@ -1,13 +1,20 @@
-import {Component} from "solid-js";
+import {Component, JSX, splitProps} from "solid-js";
+import {cx} from "../utils";
 
-interface Props {
+interface Props extends JSX.HTMLAttributes<HTMLSpanElement> {
   text?: string;
   capitalize?: boolean;
 }
 
 /** Displays a span with the specified text with its first letter capitalised using CSS. */
-export const Capitalize: Component<Props> = (props) => (
-  <span class="inline-block" classList={{"first-letter:capitalize": props.capitalize ?? true}}>
-    {props.text}
-  </span>
-);
+export const Capitalize: Component<Props> = (props) => {
+  const [localProps, spanProps] = splitProps(props, ["text", "capitalize"]);
+  return (
+    <span
+      {...spanProps}
+      class={cx(spanProps.class, "inline-block", (localProps.capitalize ?? true) && "first-letter:capitalize")}
```

Comment:
mogło zostać
```ts
cx(spanProps.class, "inline-block", {"first-letter:capitalize": localProps.capitalize ?? true })
```
ale to nie ma znaczenia a wręcz Twoje ekonomiczniejsze, możesz zignorować

Reply:
Done.